commons/src/store/files.ts

```diff
@@ -0,0 +1,52 @@
+import { writable } from "svelte/store";
+import type { File, Message } from "@commons/types/Nylas";
+import { downloadFile } from "@commons/connections/files";
+
+function initializeFilesForMessage() {
+  const { subscribe, set, update } = writable<
+    Record<string, Record<string, File>>
+  >({});
+  const filesMap: Record<string, Record<string, File>> = {}; // {message_id: {file_id: Base64 String}}
+
+  return {
+    subscribe,
+    getFilesForMessage: async (
+      incomingMessage: Message,
+      query: { component_id: string; access_token: string },
+    ) => {
+      if (!filesMap[incomingMessage.id]) {
+        const inlineFiles: Record<string, File> = {};
+        for (const file of incomingMessage.files.values()) {
+          if (file.content_disposition === "inline") {
+            if (!inlineFiles[file.id]) {
+              inlineFiles[file.id] = file;
+              inlineFiles[file.id].data = await downloadFile({
```

Comment:
This seems fine since you mentioned that it's non-blocking, but I wonder if we could improve it by storing all of the promises in an array / map and then `Promise.all`ing them. Not sure how much that would improve things here though. cc @AaronDDM for your thoughts.

Reply:
I guess if we switched to a Promise.all() scenario, we can make multiple simultaneous requests. That way if we started with a large file, it wouldn't block the smaller subsequent files from being downloaded.

Honestly though, small gains and depends on how many and how big these inline images are. I wouldn't worry about it.

Reply:
@ozsivanov `Promise.all` will reject if one of the requests fail. May be we should consider using `Promise.allSettled` instead?